### DISCUSSION TITLE
How to unsubscribe a subscription using graphql-ws with apollo client

### DISCUSSION THREAD

**Discussion by edwardfeng-db** [enisdenjo/graphql-ws#132] [Q&A] [answered]
I'm setting up the websocket client following [client usage with Apollo](https://github.com/enisdenjo/graphql-ws#apollo-client). And I'm trying to use Apollo's [subscribeToMore ](https://www.apollographql.com/docs/react/data/subscriptions/#subscribing-to-updates-for-a-query)functionality, like the following:

```
useEffect(() => {
    const unsubscribe = subscribeToMore(/*...*/)
    return () => unsubscribe();
}, [])

```
However, calling unsubscribe doesn't seem to send the `Complete` message to the server. I'm wondering why is that and do I need anything else when setting up the client? And in general how can I send the `Complete` message to the server with apollo client + graphql-ws? Thanks!

**Comment by enisdenjo** [comment, marked as answer]
Hey there, the `Complete` message is sent when the subscription is done with. You dont explicitly send the message, you `unsubscribe` from a subscription.

```js
const unsubscribe = client.subscribe(
  {
    query: 'subscription { greetings }',
  },
  {
    next: onNext,
    error: onError,
    complete: onComplete,
  },
);

// calling `unsubscribe` will send a `Complete` message for the subscription
unsubscribe();
```

Regarding your question about `subscribeToMore` - I actually have no idea about its inner workings (and I also have 0 experience with the Apollo stack), I am sorry, but I cannot be of any help there. 😬

**Reply by edwardfeng-db** [reply]
It was actually a bug on my end with the setup of apollo link. Finally found the cause! Thank you so much! @enisdenjo

**Reply by enisdenjo** [reply]
Awesome, you're very welcome!
